File: src/main/org/nlogo/prim/etc/_pwd.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim.etc

import org.nlogo.api.{ Syntax, OutputDestination }
import org.nlogo.nvm.{ Command, Context }

class _pwd extends Command {
  override def syntax =
    Syntax.commandSyntax("O---", false)
  override def perform(context: Context) {
    val path =
      Option(workspace.getModelPath)
        .getOrElse("no model loaded!")
    workspace.outputObject(path, null, true, true,
      OutputDestination.Normal)
    context.ip = next
  }
}

File: netlogo-gui/src/main/prim/etc/_pwd.java
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim.etc;

import org.nlogo.api.LogoException;
import org.nlogo.api.OutputDestinationJ;
import org.nlogo.api.Syntax;
import org.nlogo.nvm.Command;

/**
 * prints the working directory
 */
public final strictfp class _pwd
    extends Command {
  @Override
  public org.nlogo.core.Syntax syntax() {
    return Syntax.commandSyntax("O---");
  }

  @Override
  public void perform(final org.nlogo.nvm.Context context)
      throws LogoException {
    String path = workspace.getModelPath();
    if (path == null) {
      path = "no model loaded!";
    }
    workspace.outputObject
        (path, null, true, true,
            OutputDestinationJ.NORMAL());
    context.ip = next;
  }
}
